src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -161,10 +161,9 @@ private void EnsureHashExists(FileData file)
             }
             
             var hasSha256Hash = file.Hashes.Any(x => x.Algorithm == AlgorithmKind.Sha256);
-            if (!hasSha256Hash)
+            if (!hasSha256Hash && !string.IsNullOrWhiteSpace(file.Contents.Binary))
```

Comment:
But now we never compute the hash of a file for which we only have text contents. And in fact we _can't_ do that in general without knowing the encoding of the original file.

And in fact the computation of the hash for Binary is wrong. For now, change the logic to "if you have Contents.Text, take the hash of that; otherwise do nothing". Then let's merge that, and I'll call you to talk about what to do next.